enterprise/frontend/src/embedding-sdk/lib/usage-problem.ts

```diff
@@ -61,43 +81,58 @@ export function getSdkUsageProblem(
       hasJwtProviderUriProperty,
     })
       .with({ hasJwtProviderUriProperty: true }, () =>
-        toError(PROBLEMS.JWT_PROVIDER_URI_DEPRECATED),
+        toError("JWT_PROVIDER_URI_DEPRECATED"),
       )
       .with({ isSSO: false, isApiKey: false }, () =>
-        toError(PROBLEMS.NO_AUTH_METHOD_PROVIDED),
+        toError("NO_AUTH_METHOD_PROVIDED"),
       )
       .with({ isSSO: true, isApiKey: true }, () =>
-        toError(PROBLEMS.CONFLICTING_AUTH_METHODS),
+        toError("CONFLICTING_AUTH_METHODS"),
       )
       // For SSO, the token features and the toggle must both be enabled.
       .with({ isSSO: true, hasTokenFeature: true, isEnabled: true }, () => null)
+      // We cannot detect if embedding is disabled on non-localhost environments,
+      // as CORS is disabled on /api/session/properties.
+      .with(
+        {
+          isSSO: true,
+          hasTokenFeature: true,
+          isLocalhost: true,
+          isEnabled: false,
+        },
+        () => toError("EMBEDDING_SDK_NOT_ENABLED"),
+      )
```

Comment:
If you are in non-localhost domains (e.g. `https://example.com`) and the sdk embedding settings is disabled, CORS would have also been disabled for Metabase APIs, therefore we would not be able to query `/api/session/properties` to get the state of `settings["enable-embedding-sdk"]`, therefore this check would **not** work.

You could remove `isLocalhost: true` here in this matcher and it would've been fine, but I decided to add it here to be more explicit that this check only works on localhost in reality.